src/DotVVM.Framework/Resources/Scripts/DotVVM.Validation.ts

```diff
@@ -304,6 +305,15 @@ class DotvvmValidation {
                         item.innerText = error.errorMessage;
                         element.appendChild(item);
                     }
+
+                    if (binding.hideWhenValid == true) {
```

Comment:
I'd just treat all truthy values as `true` (as we do for `includeErrorsFromChildren`, ...)
```suggestion
                    if (binding.hideWhenValid) {
```

or, if you would really want to accept only `true`, then you must use `===`.
```suggestion
                    if (binding.hideWhenValid === true) {
```